logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/MessageSender.kt

```diff
@@ -145,15 +146,23 @@ internal class MessageSenderImpl internal constructor(
 
                 is Conversation.ProtocolInfo.Proteus -> {
                     // TODO(messaging): make this thread safe (per user)
-                    attemptToSendWithProteus(message)
+                    attemptToSendWithProteus(message, targetRecipients)
                 }
             }
         }
-}
+    }
 
-    private suspend fun attemptToSendWithProteus(message: Message.Regular): Either<CoreFailure, String> {
+    private suspend fun attemptToSendWithProteus(
+        message: Message.Regular,
+        targetRecipients: List<Recipient>?
```

Comment:
In this case where you pass in a subset of the possible recipients we also needs modify the `trySendingProteusEnvelope`. Currently it always send the message with the `MessageApi.QualifiedMessageOption.ReportAll` option but when a partial list is sent we need to use one of the following options:

`IgnoreAll` - If the message is sent to a single client
`ReportSome` = if the message is sent to a single user

Reply:
Might make sense to create an enum to make intention clear instead of trying to compute it.

```
sealed class MessageTarget {
  class User(val userId: UserId): MessageTarget
  class Client(val clientId: ClientId): MessageTarget
  class Conversation: MesageTarget
}
```